### PR TITLE
PARQUET-434: Add a ParquetFileReader class

### PR DESCRIPTION
**Pull request by wesm**
Starting point for build out.

### CODE REVIEW

**Comment by wesm** [comment]
@nongli this is ready for review. Requires #19

**Comment by wesm** [comment]
I'm going to squash these commits and also #19 to make cherry-picking easier for me

**Comment by wesm** [comment]
@nongli 

- Got rid of that `new` in favor of `std::vector`
- The `Close` / dtor stuff in `ParquetFileReader` has been updated in PARQUET-451 https://github.com/wesm/parquet-cpp/commit/e18b84775eb87630209ac39c1bde1263c68e45e4#diff-57e387c69a2c44010d45f314706f87b8R44

**Comment by wesm** [comment]
Rebasing on master

**Comment by wesm** [comment]
@nongli this is ready for merge

**Comment by wesm** [comment]
PARQUET-449 introduced a Thrift conflict with googletest on Linux / gcc-4.9 that I'm sorting out. Am able to repro locally, will push a fix

**Comment by wesm** [comment]
I'm going to rebase this on #25 so we can get a passing build

**Comment by wesm** [comment]
OK, back to a passing build on thrift 0.9.0 generated sources 

https://travis-ci.org/wesm/parquet-cpp/builds/104997061

**Comment by wesm** [comment]
This is rebased on master and ready to go

File: src/parquet/reader-test.cc [hunk 1]
```diff
@@ -15,12 +15,45 @@
...
 #include <gtest/gtest.h>
 
-namespace parquet {
+#include "parquet/reader.h"
+
+using std::string;
+
+namespace parquet_cpp {
+
+const char* data_dir = std::getenv("PARQUET_TEST_DATA");
```
Comment (nongli) [inline src/parquet/reader-test.cc]: This should be documented or ideally configured by the set up script. It also just segfaults if it is not set which is not ideal.

Comment (wesm) [inline src/parquet/reader-test.cc]: Good point. I'll add an entry to `setup_build_env.sh` and add something to the README

Comment (wesm) [inline src/parquet/reader-test.cc]: done

**Comment by nongli** [comment]
LGTM pending travis.

We should try to make the env set up script the same as the travis script eventually

**Comment by wesm** [comment]
Sounds good. I'll open a JIRA

**Comment by wesm** [comment]
https://issues.apache.org/jira/browse/PARQUET-471

**Comment by wesm** [comment]
here is a verified build for 8a9e875

https://travis-ci.org/wesm/parquet-cpp/builds/105024213

**Comment by nongli** [comment]
merged. Thanks!

**Comment by wesm** [comment]
thanks! rebasing #23  now

**Comment by wesm** [comment]
Er actually I will wait for #18 to be merged

**Pull request closed without merging**